garrysmod/gamemodes/terrortown/gamemode/vgui/sb_team.lua

```diff
@@ -116,11 +101,41 @@ end
 
 function PANEL:UpdateSortCache()
    self.rows_sorted = {}
-   for k,v in pairs(self.rows) do
-      table.insert(self.rows_sorted, v)
+
+   for _, row in pairs(self.rows) do
+      table.insert(self.rows_sorted, row)
    end
 
-   table.sort(self.rows_sorted, CompareScore)
+   table.sort(self.rows_sorted, function(rowa, rowb)
+      local plya = rowa:GetPlayer()
+      local plyb = rowb:GetPlayer()
+
+      if not IsValid(plya) then return false end
+      if not IsValid(plyb) then return true end
+
+      local comp = 0 -- Lua doesnt have an Ordering enumeration, I think?
+      
+      local sort_mode = GetConVar("ttt_scoreboard_sorting"):GetString()
+      local sort_func = sboard_panel.sort_table[sort_mode]

```

Comment:
This use of `sboard_panel` breaks things. `UpdateSortCache` is called while initializing the scoreboard. At that point, `sboard_panel` is nil (because the panel is still being created).

As a result, if 2+ players exist when you first open the scoreboard, then this `sboard_panel.sort_table` access will of course break and throw your face full of errors.

Note that simply not sorting if `sboard_panel == nil` probably will not work, because it will just leave the sort cache in an unsorted state until something triggers an update.